zebra-chain/src/transaction.rs

```diff
@@ -540,4 +542,147 @@ impl Transaction {
+                joinsplit_data,
+                ..
+            } => {
+                let transparent = transparent_value_pool(inputs, outputs, utxos);
+                let sprout = sprout_value_pool(joinsplit_data);
+
+                let mut value_balance = ValueBalance::zero();
+                value_balance.set_transparent_value_balance(transparent?);
+                value_balance.set_sprout_value_balance(sprout?);
+                Ok(value_balance)
+            }
+            Transaction::V3 {
+                inputs,
+                outputs,
+                joinsplit_data,
+                ..
+            } => {
+                let transparent = transparent_value_pool(inputs, outputs, utxos);
+                let sprout = sprout_value_pool(joinsplit_data);
+
+                let mut value_balance = ValueBalance::zero();
+                value_balance.set_transparent_value_balance(transparent?);
+                value_balance.set_sprout_value_balance(sprout?);
+                Ok(value_balance)
+            }
+            Transaction::V4 {
+                inputs,
+                outputs,
+                joinsplit_data,
+                sapling_shielded_data,
+                ..
+            } => {
+                let transparent = transparent_value_pool(inputs, outputs, utxos);
+                let sprout = sprout_value_pool(joinsplit_data);
+                let sapling = sapling_value_pool(sapling_shielded_data);
+
+                let mut value_balance = ValueBalance::zero();
+                value_balance.set_transparent_value_balance(transparent?);
+                value_balance.set_sprout_value_balance(sprout?);
+                value_balance.set_sapling_value_balance(sapling?);
+                Ok(value_balance)
+            }
+            Transaction::V5 {
+                inputs,
+                outputs,
+                sapling_shielded_data,
+                orchard_shielded_data,
+                ..
+            } => {
+                let transparent = transparent_value_pool(inputs, outputs, utxos);
+                let sapling = sapling_value_pool(sapling_shielded_data);
+                let orchard = orchard_value_pool(orchard_shielded_data);
+
+                let mut value_balance = ValueBalance::zero();
+                value_balance.set_transparent_value_balance(transparent?);
+                value_balance.set_sapling_value_balance(sapling?);
+                value_balance.set_orchard_value_balance(orchard?);
+                Ok(value_balance)
+            }
+        }
+    }
+}
+
+// Value pool utility functions
+
+/// Return the transparent value pool
+fn transparent_value_pool(
+    inputs: &[transparent::Input],
+    outputs: &[transparent::Output],
+    utxos: &HashMap<transparent::OutPoint, transparent::OrderedUtxo>,
+) -> Result<ValueBalance<NegativeAllowed>, Box<dyn std::error::Error>> {
+    if inputs.is_empty() && outputs.is_empty() {
+        return Ok(ValueBalance::zero());
+    }
```

Comment:
Do we need special handling for this case, or does the rest of the code produce the same result anyway?